libs/lensfun/lens.cpp

```diff
@@ -635,11 +624,11 @@ const char *lfLens::GetLensTypeDesc (lfLensType type, const char **details)
     return NULL;
 }
 
-lfLensCalibrationSet* lfLens::GetClosestCalibrationSet(const float crop) const
+std::shared_ptr<lfLensCalibrationSet> lfLens::GetClosestCalibrationSet(const float crop) const
```

Comment:
You are correct, I used the shared_ptr because it could be

Reply:
I think I will keep the shared_ptr for now, because some refactoring is needed before we can replace it with a unique_ptr, some functions return a pointer to a object in a list, they need to be refactored. We shouldn't create a new object just to return it.

Reply:
If we are talking about a pointer ownership and lifetime duration - we don't need to refactor most of the code to use `unique_ptr`, because we can keep most of the raw pointers as they were.

In the current state `lfLens` owns the pointer, it cares about a memory management and so on. When the pointer goes outside of the class, it's just a pointer and all the users of that pointer expect only two things:
1) they don't manage the pointer
2) the pointer will be alive for the whole duration
(I guess you can't do any other default assumptions while getting a raw pointer. Any other behavior must be explicitly documented)
In C++ with the smart pointers you have two options - `unique_ptr` for the default case, and `shared_ptr` for the cases where we want to share pointer ownership.
For the `unique_ptr` you move ownership to the smart pointer and expose the object as a raw pointer with those two default assumptions.
For the `shared_ptr` you can get rid of those assumptions, share the ownership and don't care about the pointer duration. That could be useful, but it's not a common case.
Usually (and especially for legacy code based on raw pointer) a `unique_ptr` + non-managed raw pointers are needed.